### PR TITLE
[PINOT-3310]: Added test to check older star-tree index can be read in and answer queries correctly (Approver: kishoreg).

### PR DESCRIPTION
**Pull request by mayankshriv**
With changes to star-tree data-structure, we need to be able to read in
older segments, and answer queries correctly. The newly added test
ensures the same.
1. Added new test as described above.
2. Renamed StarTreeIntegrationTest -> StarTreeIndexTest
3. Refactored most common code from StarTreeIndexTest into base class to
be able to resue it in the new test.

### CODE REVIEW

**Comment by kishoreg** [comment]
was the binary file addition intentional?

File: pinot-core/src/test/java/com/linkedin/pinot/core/startree/BaseStarTreeIndexTest.java [hunk 1]
```diff
@@ -42,95 +42,64 @@
...
-        "select sum(m1) from T where d1 in ('d1-v1', 'd1-v2') and d2 not in ('d2-v1') group by d3",
-        "select sum(m1) from T where d1 in ('d1-v1', 'd1-v2') and d2 not in ('d2-v1') group by d3, d4"
-    };
-
+  protected final long _randomSeed = System.nanoTime();
+
+
+  protected String[] _hardCodedQueries =
+      new String[]{
+          "select sum(m1) from T",
```
Comment (kishoreg) [inline pinot-core/src/test/java/com/linkedin/pinot/core/startree/BaseStarTreeIndexTest.java]: add a select count(*) for sanity check

**Comment by kishoreg** [comment]
LGTM.

File: pinot-core/src/test/java/com/linkedin/pinot/core/startree/BaseStarTreeIndexTest.java (hunk 1 above)
Comment (mayankshriv) [inline pinot-core/src/test/java/com/linkedin/pinot/core/startree/BaseStarTreeIndexTest.java]: ATM, the test simply aggregates on all metric columns, without looking at the actual aggregations in the query. This was done to be more exhaustive and perform aggregations on all metrics, without needing to explicitly write-down those queries. Will handle count(*) in upcoming PR's.

**Pull request merged**